src/cryptsetup/cryptsetup-generator.c

```diff
@@ -169,6 +169,21 @@ static int create_disk(
         tmp = fstab_test_option(options, "tmp\0");
         swap = fstab_test_option(options, "swap\0");
         netdev = fstab_test_option(options, "_netdev\0");
+        trypassword = fstab_test_option(options, "trypassword\0");
+
+        c = strrchr(keyspec, ':');
+        if (c) {
+            *c = '\0';
```

Comment:
keyspec is `const char*`, you are patching around in a constant string hence. Do not do that, plase.